Firestore/src/Filter.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Firestore;
+
+use Google\Cloud\Firestore\V1\StructuredQuery\CompositeFilter\Operator;
+
+/**
+ * A Query Filter class.
+ *
+ * This class helps the user to create filters for complex queries.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Firestore\FirestoreClient;
+ * use Google\Cloud\Firestore\Filter;
+ *
+ * $firestore = new FirestoreClient();
+ *
+ * $collection = $firestore->collection('users');
+ * $query = $collection->where(Filter::condition('age', '>', 18));
```

Comment:
Add usage of `and` and `or` filters and implement a snippet test

Reply:
Done! Thank you!